Darwin build failure: elmPackage.elm

### Steps To Reproduce

elm fails to build on MacOS (m2) - something related to how processes are spawned. The build fails with the message "ghc-9.6.4: security: createProcess: posix_spawnp: does not exist (No such file or directory)".

Steps to reproduce the behavior:
1. nix build nixpkgs#elmPackages.elm

### Build log

```
@nix { "action": "setPhase", "phase": "setupCompilerEnvironmentPhase" }
Running phase: setupCompilerEnvironmentPhase
Build with /nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4.
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/nxjfhirisx1cjd19kzdfcgx43x7m4xm6-compiler-2f6dd29
source root is compiler-2f6dd29
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "compileBuildDriverPhase" }
Running phase: compileBuildDriverPhase
setupCompileFlags: -package-db=/private/tmp/nix-build-elm-0.19.1.drv-0/tmp.4QNVsI6CYs/setup-package.conf.d -threaded -rtsopts
[1 of 2] Compiling Main             ( /nix/store/4mdp8nhyfddh7bllbi7xszz7k9955n79-Setup.hs, /private/tmp/nix-build-elm-0.19.1.drv-0/tmp.4QNVsI6CYs/Main.o )
[2 of 2] Linking Setup
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
configureFlags: --verbose --prefix=/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1 --libdir=$prefix/lib/$compiler/lib --libsubdir=$abi/$libname --docdir=/nix/store/mv9pnhxvlbkf87jfy1qb3h92ln4z3vgb-elm-0.19.1-doc/share/doc/elm-0.19.1 --with-gcc=clang --package-db=/private/tmp/nix-build-elm-0.19.1.drv-0/tmp.4QNVsI6CYs/package.conf.d --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --disable-split-sections --enable-library-stripping --enable-executable-stripping --extra-lib-dirs=/nix/store/iz66rbm4c48m920zjiic3sc8ghl8pvgs-ncurses-6.4/lib --extra-lib-dirs=/nix/store/0z3mx40zc1b2zf8yjlrkk0iyps3yxlil-libffi-3.4.6/lib --extra-lib-dirs=/nix/store/5kg229dp8nb6a2m0hv7g1zl725kxp0nn-gmp-with-cxx-6.3.0/lib --extra-include-dirs=/nix/store/n7hyb4xfwjdcm87gbisck8pl9c2mp1qs-libiconv-50/include --extra-lib-dirs=/nix/store/n7hyb4xfwjdcm87gbisck8pl9c2mp1qs-libiconv-50/lib --extra-include-dirs=/nix/store/rjgv94p1nnj02v0f9r2jpann9x02b4vw-libcxx-16.0.6-dev/include --extra-lib-dirs=/nix/store/j3yw5w8pbdv4argnmx7i4i0lvglnh7ll-libcxx-16.0.6/lib --extra-include-dirs=/nix/store/s6hsrbxn6wfws552x9jbn4agjz71ps3x-libcxxabi-16.0.6-dev/include --extra-lib-dirs=/nix/store/g9x4r8l8hjk3vfj26k90p3rjc6g8j3i6-libcxxabi-16.0.6/lib --extra-include-dirs=/nix/store/rbjxdnzf9kc1in1n1ca7phddg80h55xd-compiler-rt-libc-16.0.6-dev/include --extra-lib-dirs=/nix/store/v7rhm5d36dn6bgcbdc4w3dg8ym3vxb5d-compiler-rt-libc-16.0.6/lib --extra-lib-dirs=/nix/store/7y2jn8rdygiy22rzy66wsz4gvrfn5hh6-apple-framework-CoreFoundation-11.0.0/lib --extra-framework-dirs=/nix/store/7y2jn8rdygiy22rzy66wsz4gvrfn5hh6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks --extra-include-dirs=/nix/store/cii11dfrz126yb3d03aqyahwq3fanl1g-libobjc-11.0.0/include --extra-lib-dirs=/nix/store/cii11dfrz126yb3d03aqyahwq3fanl1g-libobjc-11.0.0/lib
Using Parsec parser
Configuring elm-0.19.1...
Flags chosen: dev=False
Warning: Packages using 'cabal-version: >= 1.10' and before 'cabal-version:
3.4' must specify the 'default-language' field for each component (e.g.
Haskell98 or Haskell2010). If a component uses different languages in
different modules then list the other ones in the 'other-languages' field.
Dependency HTTP: using HTTP-4000.4.1
Dependency SHA: using SHA-1.6.4.4
Dependency ansi-terminal: using ansi-terminal-1.0.2
Dependency ansi-wl-pprint <1: using ansi-wl-pprint-0.6.8.1
Dependency base: using base-4.18.2.0
Dependency binary: using binary-0.8.9.1
Dependency bytestring: using bytestring-0.11.5.3
Dependency containers: using containers-0.6.7
Dependency directory: using directory-1.3.8.1
Dependency edit-distance: using edit-distance-0.2.2.1
Dependency file-embed: using file-embed-0.0.16.0
Dependency filelock: using filelock-0.1.1.7
Dependency filepath: using filepath-1.4.200.1
Dependency ghc-prim: using ghc-prim-0.10.0
Dependency haskeline: using haskeline-0.8.2.1
Dependency http-client: using http-client-0.7.16
Dependency http-client-tls: using http-client-tls-0.3.6.3
Dependency http-types: using http-types-0.12.4
Dependency language-glsl: using language-glsl-0.3.0
Dependency mtl: using mtl-2.3.1
Dependency network: using network-3.1.4.0
Dependency parsec: using parsec-3.1.16.1
Dependency process: using process-1.6.17.0
[production] Thu Mar 21 10:26:24:~/Source/tmp
$ nix log  --log-format raw nixpkgs#elmPackages.elm  | pbcopy
[production] Thu Mar 21 10:26:27:~/Source/tmp
$ nix log  --log-format raw nixpkgs#elmPackages.elm  | pbcopy
[production] Thu Mar 21 10:26:37:~/Source/tmp
$ nix log  --log-format raw nixpkgs#elmPackages.elm
@nix { "action": "setPhase", "phase": "setupCompilerEnvironmentPhase" }
Running phase: setupCompilerEnvironmentPhase
Build with /nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4.
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/nxjfhirisx1cjd19kzdfcgx43x7m4xm6-compiler-2f6dd29
source root is compiler-2f6dd29
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "compileBuildDriverPhase" }
Running phase: compileBuildDriverPhase
setupCompileFlags: -package-db=/private/tmp/nix-build-elm-0.19.1.drv-0/tmp.4QNVsI6CYs/setup-package.conf.d -threaded -rtsopts
[1 of 2] Compiling Main             ( /nix/store/4mdp8nhyfddh7bllbi7xszz7k9955n79-Setup.hs, /private/tmp/nix-build-elm-0.19.1.drv-0/tmp.4QNVsI6CYs/Main.o )
[2 of 2] Linking Setup
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
configureFlags: --verbose --prefix=/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1 --libdir=$prefix/lib/$compiler/lib --libsubdir=$abi/$libname --docdir=/nix/store/mv9pnhxvlbkf87jfy1qb3h92ln4z3vgb-elm-0.19.1-doc/share/doc/elm-0.19.1 --with-gcc=clang --package-db=/private/tmp/nix-build-elm-0.19.1.drv-0/tmp.4QNVsI6CYs/package.conf.d --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --disable-split-sections --enable-library-stripping --enable-executable-stripping --extra-lib-dirs=/nix/store/iz66rbm4c48m920zjiic3sc8ghl8pvgs-ncurses-6.4/lib --extra-lib-dirs=/nix/store/0z3mx40zc1b2zf8yjlrkk0iyps3yxlil-libffi-3.4.6/lib --extra-lib-dirs=/nix/store/5kg229dp8nb6a2m0hv7g1zl725kxp0nn-gmp-with-cxx-6.3.0/lib --extra-include-dirs=/nix/store/n7hyb4xfwjdcm87gbisck8pl9c2mp1qs-libiconv-50/include --extra-lib-dirs=/nix/store/n7hyb4xfwjdcm87gbisck8pl9c2mp1qs-libiconv-50/lib --extra-include-dirs=/nix/store/rjgv94p1nnj02v0f9r2jpann9x02b4vw-libcxx-16.0.6-dev/include --extra-lib-dirs=/nix/store/j3yw5w8pbdv4argnmx7i4i0lvglnh7ll-libcxx-16.0.6/lib --extra-include-dirs=/nix/store/s6hsrbxn6wfws552x9jbn4agjz71ps3x-libcxxabi-16.0.6-dev/include --extra-lib-dirs=/nix/store/g9x4r8l8hjk3vfj26k90p3rjc6g8j3i6-libcxxabi-16.0.6/lib --extra-include-dirs=/nix/store/rbjxdnzf9kc1in1n1ca7phddg80h55xd-compiler-rt-libc-16.0.6-dev/include --extra-lib-dirs=/nix/store/v7rhm5d36dn6bgcbdc4w3dg8ym3vxb5d-compiler-rt-libc-16.0.6/lib --extra-lib-dirs=/nix/store/7y2jn8rdygiy22rzy66wsz4gvrfn5hh6-apple-framework-CoreFoundation-11.0.0/lib --extra-framework-dirs=/nix/store/7y2jn8rdygiy22rzy66wsz4gvrfn5hh6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks --extra-include-dirs=/nix/store/cii11dfrz126yb3d03aqyahwq3fanl1g-libobjc-11.0.0/include --extra-lib-dirs=/nix/store/cii11dfrz126yb3d03aqyahwq3fanl1g-libobjc-11.0.0/lib
Using Parsec parser
Configuring elm-0.19.1...
Flags chosen: dev=False
Warning: Packages using 'cabal-version: >= 1.10' and before 'cabal-version:
3.4' must specify the 'default-language' field for each component (e.g.
Haskell98 or Haskell2010). If a component uses different languages in
different modules then list the other ones in the 'other-languages' field.
Dependency HTTP: using HTTP-4000.4.1
Dependency SHA: using SHA-1.6.4.4
Dependency ansi-terminal: using ansi-terminal-1.0.2
Dependency ansi-wl-pprint <1: using ansi-wl-pprint-0.6.8.1
Dependency base: using base-4.18.2.0
Dependency binary: using binary-0.8.9.1
Dependency bytestring: using bytestring-0.11.5.3
Dependency containers: using containers-0.6.7
Dependency directory: using directory-1.3.8.1
Dependency edit-distance: using edit-distance-0.2.2.1
Dependency file-embed: using file-embed-0.0.16.0
Dependency filelock: using filelock-0.1.1.7
Dependency filepath: using filepath-1.4.200.1
Dependency ghc-prim: using ghc-prim-0.10.0
Dependency haskeline: using haskeline-0.8.2.1
Dependency http-client: using http-client-0.7.16
Dependency http-client-tls: using http-client-tls-0.3.6.3
Dependency http-types: using http-types-0.12.4
Dependency language-glsl: using language-glsl-0.3.0
Dependency mtl: using mtl-2.3.1
Dependency network: using network-3.1.4.0
Dependency parsec: using parsec-3.1.16.1
Dependency process: using process-1.6.17.0
Dependency raw-strings-qq: using raw-strings-qq-1.1
Dependency scientific: using scientific-0.3.7.0
Dependency snap-core: using snap-core-1.0.5.1
Dependency snap-server: using snap-server-1.1.2.1
Dependency template-haskell: using template-haskell-2.20.0.0
Dependency time: using time-1.12.2
Dependency unordered-containers: using unordered-containers-0.2.20
Dependency utf8-string: using utf8-string-1.0.2
Dependency vector: using vector-0.13.1.0
Dependency zip-archive: using zip-archive-0.4.3
Source component graph: component exe:elm
Configured component graph:
    component elm-0.19.1-M76nVJ6ymgCk5LWdohv4Q-elm
        include HTTP-4000.4.1-FiFXQHyiMAu228LSgpXrxx
        include SHA-1.6.4.4-2Cv9bt0oImGj03HsSWVLG
        include ansi-terminal-1.0.2-GmRyinTsbXe6lY3Bm6ersV
        include ansi-wl-pprint-0.6.8.1-Ziodk3wXIo2FCPjThn76t
        include base-4.18.2.0
        include binary-0.8.9.1
        include bytestring-0.11.5.3
        include containers-0.6.7
        include directory-1.3.8.1
        include edit-distance-0.2.2.1-Li5GGR1Mtm243t6RpgjV0c
        include file-embed-0.0.16.0-1I6wWqQCRNX7Jj3KH7fJB2
        include filelock-0.1.1.7-LirM8KrMtzD7RkBS14Fn0V
        include filepath-1.4.200.1
        include ghc-prim-0.10.0
        include haskeline-0.8.2.1
        include http-client-0.7.16-FJEqqStJ2brEUHy1YX5ln2
        include http-client-tls-0.3.6.3-JBW7lu1DI6hJQQPtXwlNBD
        include http-types-0.12.4-9HmHkfzXhyx9D4yTlLkyPq
        include language-glsl-0.3.0-LXtmuwHtf284sAkzKaNExG
        include mtl-2.3.1
        include network-3.1.4.0-AJEoDmw4KkzB0MEWVViGHB
        include parsec-3.1.16.1
        include process-1.6.17.0
        include raw-strings-qq-1.1-EhcRjbnc1JG73WHi6l6OFo
        include scientific-0.3.7.0-EvqtudccFkO85gGnrSbJyn
        include snap-core-1.0.5.1-6wiiQ69tXDYC4kosIBJMuE
        include snap-server-1.1.2.1-2canCqZNKJi5pKZJKXlmL0
        include template-haskell-2.20.0.0
        include time-1.12.2
        include unordered-containers-0.2.20-AZR8CkxJ5097lzL4grBXMW
        include utf8-string-1.0.2-EGSxFFMtDUDnpUJmf8yA8
        include vector-0.13.1.0-EKmVc2NKwg4DFMuTqZMRHr
        include zip-archive-0.4.3-D62Bih3oGAE2Dc10RYk5sj
Linked component graph:
    unit elm-0.19.1-M76nVJ6ymgCk5LWdohv4Q-elm
        include HTTP-4000.4.1-FiFXQHyiMAu228LSgpXrxx
        include SHA-1.6.4.4-2Cv9bt0oImGj03HsSWVLG
        include ansi-terminal-1.0.2-GmRyinTsbXe6lY3Bm6ersV
        include ansi-wl-pprint-0.6.8.1-Ziodk3wXIo2FCPjThn76t
        include base-4.18.2.0
        include binary-0.8.9.1
        include bytestring-0.11.5.3
        include containers-0.6.7
        include directory-1.3.8.1
        include edit-distance-0.2.2.1-Li5GGR1Mtm243t6RpgjV0c
        include file-embed-0.0.16.0-1I6wWqQCRNX7Jj3KH7fJB2
        include filelock-0.1.1.7-LirM8KrMtzD7RkBS14Fn0V
        include filepath-1.4.200.1
        include ghc-prim-0.10.0
        include haskeline-0.8.2.1
        include http-client-0.7.16-FJEqqStJ2brEUHy1YX5ln2
        include http-client-tls-0.3.6.3-JBW7lu1DI6hJQQPtXwlNBD
        include http-types-0.12.4-9HmHkfzXhyx9D4yTlLkyPq
        include language-glsl-0.3.0-LXtmuwHtf284sAkzKaNExG
        include mtl-2.3.1
        include network-3.1.4.0-AJEoDmw4KkzB0MEWVViGHB
        include parsec-3.1.16.1
        include process-1.6.17.0
        include raw-strings-qq-1.1-EhcRjbnc1JG73WHi6l6OFo
        include scientific-0.3.7.0-EvqtudccFkO85gGnrSbJyn
        include snap-core-1.0.5.1-6wiiQ69tXDYC4kosIBJMuE
        include snap-server-1.1.2.1-2canCqZNKJi5pKZJKXlmL0
        include template-haskell-2.20.0.0
        include time-1.12.2
        include unordered-containers-0.2.20-AZR8CkxJ5097lzL4grBXMW
        include utf8-string-1.0.2-EGSxFFMtDUDnpUJmf8yA8
        include vector-0.13.1.0-EKmVc2NKwg4DFMuTqZMRHr
        include zip-archive-0.4.3-D62Bih3oGAE2Dc10RYk5sj
Ready component graph:
    definite elm-0.19.1-M76nVJ6ymgCk5LWdohv4Q-elm
        depends HTTP-4000.4.1-FiFXQHyiMAu228LSgpXrxx
        depends SHA-1.6.4.4-2Cv9bt0oImGj03HsSWVLG
        depends ansi-terminal-1.0.2-GmRyinTsbXe6lY3Bm6ersV
        depends ansi-wl-pprint-0.6.8.1-Ziodk3wXIo2FCPjThn76t
        depends base-4.18.2.0
        depends binary-0.8.9.1
        depends bytestring-0.11.5.3
        depends containers-0.6.7
        depends directory-1.3.8.1
        depends edit-distance-0.2.2.1-Li5GGR1Mtm243t6RpgjV0c
        depends file-embed-0.0.16.0-1I6wWqQCRNX7Jj3KH7fJB2
        depends filelock-0.1.1.7-LirM8KrMtzD7RkBS14Fn0V
        depends filepath-1.4.200.1
        depends ghc-prim-0.10.0
        depends haskeline-0.8.2.1
        depends http-client-0.7.16-FJEqqStJ2brEUHy1YX5ln2
        depends http-client-tls-0.3.6.3-JBW7lu1DI6hJQQPtXwlNBD
        depends http-types-0.12.4-9HmHkfzXhyx9D4yTlLkyPq
        depends language-glsl-0.3.0-LXtmuwHtf284sAkzKaNExG
        depends mtl-2.3.1
        depends network-3.1.4.0-AJEoDmw4KkzB0MEWVViGHB
        depends parsec-3.1.16.1
        depends process-1.6.17.0
        depends raw-strings-qq-1.1-EhcRjbnc1JG73WHi6l6OFo
        depends scientific-0.3.7.0-EvqtudccFkO85gGnrSbJyn
        depends snap-core-1.0.5.1-6wiiQ69tXDYC4kosIBJMuE
        depends snap-server-1.1.2.1-2canCqZNKJi5pKZJKXlmL0
        depends template-haskell-2.20.0.0
        depends time-1.12.2
        depends unordered-containers-0.2.20-AZR8CkxJ5097lzL4grBXMW
        depends utf8-string-1.0.2-EGSxFFMtDUDnpUJmf8yA8
        depends vector-0.13.1.0-EKmVc2NKwg4DFMuTqZMRHr
        depends zip-archive-0.4.3-D62Bih3oGAE2Dc10RYk5sj
Using Cabal-3.10.1.0 compiled by ghc-9.6
Using compiler: ghc-9.6.4
Using install prefix: /nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1
Executables installed in:
/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1/bin
Libraries installed in:
/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1/lib/ghc-9.6.4/lib/aarch64-osx-ghc-9.6.4/elm-0.19.1
Dynamic Libraries installed in:
/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1/lib/ghc-9.6.4/lib/aarch64-osx-ghc-9.6.4
Private executables installed in:
/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1/libexec/aarch64-osx-ghc-9.6.4/elm-0.19.1
Data files installed in:
/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1/share/aarch64-osx-ghc-9.6.4/elm-0.19.1
Documentation installed in:
/nix/store/mv9pnhxvlbkf87jfy1qb3h92ln4z3vgb-elm-0.19.1-doc/share/doc/elm-0.19.1
Configuration files installed in:
/nix/store/r8571svmzhz776xwk97m8ak9hlhjj7vj-elm-0.19.1/etc
No alex found
Using ar found on system at:
/nix/store/cq2a3n6ygvzwvd5k44xfgwin041bh9h0-cctools-binutils-darwin-16.0.6-973.0.1/bin/ar
No c2hs found
No cpphs found
No doctest found
Using gcc version 16.0.6 given by user at:
/nix/store/0m04cmy32vs1686pzjqajsh50arbv8m4-clang-wrapper-16.0.6/bin/clang
Using ghc version 9.6.4 found on system at:
/nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4/bin/ghc
Using ghc-pkg version 9.6.4 found on system at:
/nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4/bin/ghc-pkg-9.6.4
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.29.2 found on system at:
/nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4/bin/haddock-ghc-9.6.4
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.68 found on system at:
/nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4/bin/hpc-ghc-9.6.4
Using hsc2hs version 0.68.9 found on system at:
/nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4/bin/hsc2hs-ghc-9.6.4
No hscolour found
No jhc found
Using ld found on system at:
/nix/store/6ira44ghp443bg60paqnrajiny1alchs-cctools-binutils-darwin-wrapper-16.0.6-973.0.1/bin/ld
No pkg-config found
Using runghc version 9.6.4 found on system at:
/nix/store/0parfyvvzpxd2n7nfxd5jzpx9b05jzlr-ghc-9.6.4/bin/runghc-9.6.4
Using strip found on system at:
/nix/store/0m04cmy32vs1686pzjqajsh50arbv8m4-clang-wrapper-16.0.6/bin/strip
Using tar found on system at:
/nix/store/h9hn3zrjcdf2ck1fp9agis9f1d6rphcw-gnutar-1.35/bin/tar
No uhc found
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
Preprocessing executable 'elm' for elm-0.19.1..
Building executable 'elm' for elm-0.19.1..
[  1 of 132] Compiling AST.Utils.Binop  ( compiler/src/AST/Utils/Binop.hs, dist/build/elm/elm-tmp/AST/Utils/Binop.o, dist/build/elm/elm-tmp/AST/Utils/Binop.dyn_o )
[  2 of 132] Compiling Data.Bag         ( compiler/src/Data/Bag.hs, dist/build/elm/elm-tmp/Data/Bag.o, dist/build/elm/elm-tmp/Data/Bag.dyn_o )
[  3 of 132] Compiling Data.Index       ( compiler/src/Data/Index.hs, dist/build/elm/elm-tmp/Data/Index.o, dist/build/elm/elm-tmp/Data/Index.dyn_o )
[  4 of 132] Compiling Data.Map.Utils   ( compiler/src/Data/Map/Utils.hs, dist/build/elm/elm-tmp/Data/Map/Utils.o, dist/build/elm/elm-tmp/Data/Map/Utils.dyn_o )
[  5 of 132] Compiling Data.NonEmptyList ( compiler/src/Data/NonEmptyList.hs, dist/build/elm/elm-tmp/Data/NonEmptyList.o, dist/build/elm/elm-tmp/Data/NonEmptyList.dyn_o )
[  6 of 132] Compiling Data.OneOrMore   ( compiler/src/Data/OneOrMore.hs, dist/build/elm/elm-tmp/Data/OneOrMore.o, dist/build/elm/elm-tmp/Data/OneOrMore.dyn_o )
[  7 of 132] Compiling Elm.Magnitude    ( compiler/src/Elm/Magnitude.hs, dist/build/elm/elm-tmp/Elm/Magnitude.o, dist/build/elm/elm-tmp/Elm/Magnitude.dyn_o )
[  8 of 132] Compiling File             ( builder/src/File.hs, dist/build/elm/elm-tmp/File.o, dist/build/elm/elm-tmp/File.dyn_o )
[  9 of 132] Compiling BackgroundWriter ( builder/src/BackgroundWriter.hs, dist/build/elm/elm-tmp/BackgroundWriter.o, dist/build/elm/elm-tmp/BackgroundWriter.dyn_o )
[ 10 of 132] Compiling Generate.JavaScript.Functions ( compiler/src/Generate/JavaScript/Functions.hs, dist/build/elm/elm-tmp/Generate/JavaScript/Functions.o, dist/build/elm/elm-tmp/Generate/JavaScript/Functions.dyn_o )
[ 11 of 132] Compiling Paths_elm        ( dist/build/elm/autogen/Paths_elm.hs, dist/build/elm/elm-tmp/Paths_elm.o, dist/build/elm/elm-tmp/Paths_elm.dyn_o )
[ 12 of 132] Compiling Reporting.Annotation ( compiler/src/Reporting/Annotation.hs, dist/build/elm/elm-tmp/Reporting/Annotation.o, dist/build/elm/elm-tmp/Reporting/Annotation.dyn_o )
[ 13 of 132] Compiling Parse.Primitives ( compiler/src/Parse/Primitives.hs, dist/build/elm/elm-tmp/Parse/Primitives.o, dist/build/elm/elm-tmp/Parse/Primitives.dyn_o )

compiler/src/Parse/Primitives.hs:85:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘pure = return’ definition detected
    in the instance declaration for ‘Applicative (Parser x)’.
    Move definition from ‘return’ to ‘pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
   |
85 |   pure = return
   |   ^^^^^^^^^^^^^

compiler/src/Parse/Primitives.hs:181:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘return’ definition detected
    in the instance declaration for ‘Monad (Parser x)’.
    ‘return’ will eventually be removed in favour of ‘pure’
    Either remove definition for ‘return’ (recommended) or define as ‘return = pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
181 |   return value =
    |   ^^^^^^^^^^^^^^...
[ 14 of 132] Compiling Data.Utf8        ( compiler/src/Data/Utf8.hs, dist/build/elm/elm-tmp/Data/Utf8.o, dist/build/elm/elm-tmp/Data/Utf8.dyn_o )
[ 15 of 132] Compiling Elm.String       ( compiler/src/Elm/String.hs, dist/build/elm/elm-tmp/Elm/String.o, dist/build/elm/elm-tmp/Elm/String.dyn_o )
[ 16 of 132] Compiling Elm.Float        ( compiler/src/Elm/Float.hs, dist/build/elm/elm-tmp/Elm/Float.o, dist/build/elm/elm-tmp/Elm/Float.dyn_o )
[ 17 of 132] Compiling Data.Name        ( compiler/src/Data/Name.hs, dist/build/elm/elm-tmp/Data/Name.o, dist/build/elm/elm-tmp/Data/Name.dyn_o )
[ 18 of 132] Compiling Parse.Symbol     ( compiler/src/Parse/Symbol.hs, dist/build/elm/elm-tmp/Parse/Symbol.o, dist/build/elm/elm-tmp/Parse/Symbol.dyn_o )
[ 19 of 132] Compiling Json.String      ( compiler/src/Json/String.hs, dist/build/elm/elm-tmp/Json/String.o, dist/build/elm/elm-tmp/Json/String.dyn_o )
[ 20 of 132] Compiling Json.Encode      ( compiler/src/Json/Encode.hs, dist/build/elm/elm-tmp/Json/Encode.o, dist/build/elm/elm-tmp/Json/Encode.dyn_o )

compiler/src/Json/Encode.hs:31:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
31 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 21 of 132] Compiling Generate.Html    ( compiler/src/Generate/Html.hs, dist/build/elm/elm-tmp/Generate/Html.o, dist/build/elm/elm-tmp/Generate/Html.dyn_o )
[ 22 of 132] Compiling Develop.Generate.Help ( terminal/src/Develop/Generate/Help.hs, dist/build/elm/elm-tmp/Develop/Generate/Help.o, dist/build/elm/elm-tmp/Develop/Generate/Help.dyn_o )

terminal/src/Develop/Generate/Help.hs:12:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
12 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 23 of 132] Compiling AST.Utils.Shader ( compiler/src/AST/Utils/Shader.hs, dist/build/elm/elm-tmp/AST/Utils/Shader.o, dist/build/elm/elm-tmp/AST/Utils/Shader.dyn_o )
[ 24 of 132] Compiling AST.Source       ( compiler/src/AST/Source.hs, dist/build/elm/elm-tmp/AST/Source.o, dist/build/elm/elm-tmp/AST/Source.dyn_o )
[ 25 of 132] Compiling Parse.Variable   ( compiler/src/Parse/Variable.hs, dist/build/elm/elm-tmp/Parse/Variable.o, dist/build/elm/elm-tmp/Parse/Variable.dyn_o )
[ 26 of 132] Compiling Parse.Keyword    ( compiler/src/Parse/Keyword.hs, dist/build/elm/elm-tmp/Parse/Keyword.o, dist/build/elm/elm-tmp/Parse/Keyword.dyn_o )
[ 27 of 132] Compiling Json.Decode      ( compiler/src/Json/Decode.hs, dist/build/elm/elm-tmp/Json/Decode.o, dist/build/elm/elm-tmp/Json/Decode.dyn_o )

compiler/src/Json/Decode.hs:122:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘pure = return’ definition detected
    in the instance declaration for ‘Applicative (Decoder x)’.
    Move definition from ‘return’ to ‘pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
122 |   pure = return
    |   ^^^^^^^^^^^^^

compiler/src/Json/Decode.hs:139:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘return’ definition detected
    in the instance declaration for ‘Monad (Decoder x)’.
    ‘return’ will eventually be removed in favour of ‘pure’
    Either remove definition for ‘return’ (recommended) or define as ‘return = pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
139 |   return a =
    |   ^^^^^^^^^^...
[ 28 of 132] Compiling Elm.Version      ( compiler/src/Elm/Version.hs, dist/build/elm/elm-tmp/Elm/Version.o, dist/build/elm/elm-tmp/Elm/Version.dyn_o )
[ 29 of 132] Compiling Http             ( builder/src/Http.hs, dist/build/elm/elm-tmp/Http.o, dist/build/elm/elm-tmp/Http.dyn_o )
[ 30 of 132] Compiling Elm.Constraint   ( compiler/src/Elm/Constraint.hs, dist/build/elm/elm-tmp/Elm/Constraint.o, dist/build/elm/elm-tmp/Elm/Constraint.dyn_o )
[ 31 of 132] Compiling Reporting.Suggest ( compiler/src/Reporting/Suggest.hs, dist/build/elm/elm-tmp/Reporting/Suggest.o, dist/build/elm/elm-tmp/Reporting/Suggest.dyn_o )
[ 32 of 132] Compiling Elm.Package      ( compiler/src/Elm/Package.hs, dist/build/elm/elm-tmp/Elm/Package.o, dist/build/elm/elm-tmp/Elm/Package.dyn_o )

compiler/src/Elm/Package.hs:37:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
37 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 33 of 132] Compiling Reporting.Doc    ( compiler/src/Reporting/Doc.hs, dist/build/elm/elm-tmp/Reporting/Doc.o, dist/build/elm/elm-tmp/Reporting/Doc.dyn_o )

compiler/src/Reporting/Doc.hs:49:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
49 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

compiler/src/Reporting/Doc.hs:341:9: warning: [GHC-62161] [-Wincomplete-patterns]
    Pattern match(es) are non-exhaustive
    In a case alternative:
        Patterns of type ‘Ansi.SGR’ not matched:
            Ansi.SetPaletteColor _ _
            Ansi.SetDefaultColor _
    |
341 |         case sgr of
    |         ^^^^^^^^^^^...
[ 34 of 132] Compiling Reporting.Report ( compiler/src/Reporting/Report.hs, dist/build/elm/elm-tmp/Reporting/Report.o, dist/build/elm/elm-tmp/Reporting/Report.dyn_o )
[ 35 of 132] Compiling Reporting.Render.Code ( compiler/src/Reporting/Render/Code.hs, dist/build/elm/elm-tmp/Reporting/Render/Code.o, dist/build/elm/elm-tmp/Reporting/Render/Code.dyn_o )

compiler/src/Reporting/Render/Code.hs:181:7: warning: [GHC-62161] [-Wincomplete-uni-patterns]
    Pattern match(es) are non-exhaustive
    In a pattern binding:
        Patterns of type ‘Maybe String’ not matched: Nothing
    |
181 |       (Just line) = List.lookup startRow1 sourceLines
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 36 of 132] Compiling Elm.ModuleName   ( compiler/src/Elm/ModuleName.hs, dist/build/elm/elm-tmp/Elm/ModuleName.o, dist/build/elm/elm-tmp/Elm/ModuleName.dyn_o )
[ 37 of 132] Compiling Reporting.Render.Type.Localizer ( compiler/src/Reporting/Render/Type/Localizer.hs, dist/build/elm/elm-tmp/Reporting/Render/Type/Localizer.o, dist/build/elm/elm-tmp/Reporting/Render/Type/Localizer.dyn_o )

compiler/src/Reporting/Render/Type/Localizer.hs:20:1: warning: [-Wunused-imports]
    The import of ‘Reporting.Doc’ is redundant
      except perhaps to import instances from ‘Reporting.Doc’
    To import instances alone, use: import Reporting.Doc()
   |
20 | import Reporting.Doc ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 38 of 132] Compiling Reporting.Error.Syntax ( compiler/src/Reporting/Error/Syntax.hs, dist/build/elm/elm-tmp/Reporting/Error/Syntax.o, dist/build/elm/elm-tmp/Reporting/Error/Syntax.dyn_o )
[ 39 of 132] Compiling Reporting.Error.Docs ( compiler/src/Reporting/Error/Docs.hs, dist/build/elm/elm-tmp/Reporting/Error/Docs.o, dist/build/elm/elm-tmp/Reporting/Error/Docs.dyn_o )

compiler/src/Reporting/Error/Docs.hs:19:1: warning: [-Wunused-imports]
    The import of ‘Reporting.Doc’ is redundant
      except perhaps to import instances from ‘Reporting.Doc’
    To import instances alone, use: import Reporting.Doc()
   |
19 | import Reporting.Doc ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 40 of 132] Compiling Parse.Space      ( compiler/src/Parse/Space.hs, dist/build/elm/elm-tmp/Parse/Space.o, dist/build/elm/elm-tmp/Parse/Space.dyn_o )
[ 41 of 132] Compiling Parse.Type       ( compiler/src/Parse/Type.hs, dist/build/elm/elm-tmp/Parse/Type.o, dist/build/elm/elm-tmp/Parse/Type.dyn_o )
[ 42 of 132] Compiling Parse.Shader     ( compiler/src/Parse/Shader.hs, dist/build/elm/elm-tmp/Parse/Shader.o, dist/build/elm/elm-tmp/Parse/Shader.dyn_o )
[ 43 of 132] Compiling Parse.Number     ( compiler/src/Parse/Number.hs, dist/build/elm/elm-tmp/Parse/Number.o, dist/build/elm/elm-tmp/Parse/Number.dyn_o )
[ 44 of 132] Compiling Parse.String     ( compiler/src/Parse/String.hs, dist/build/elm/elm-tmp/Parse/String.o, dist/build/elm/elm-tmp/Parse/String.dyn_o )
[ 45 of 132] Compiling Parse.Pattern    ( compiler/src/Parse/Pattern.hs, dist/build/elm/elm-tmp/Parse/Pattern.o, dist/build/elm/elm-tmp/Parse/Pattern.dyn_o )
[ 46 of 132] Compiling Parse.Expression ( compiler/src/Parse/Expression.hs, dist/build/elm/elm-tmp/Parse/Expression.o, dist/build/elm/elm-tmp/Parse/Expression.dyn_o )
[ 47 of 132] Compiling Parse.Declaration ( compiler/src/Parse/Declaration.hs, dist/build/elm/elm-tmp/Parse/Declaration.o, dist/build/elm/elm-tmp/Parse/Declaration.dyn_o )
[ 48 of 132] Compiling Reporting.Error.Import ( compiler/src/Reporting/Error/Import.hs, dist/build/elm/elm-tmp/Reporting/Error/Import.o, dist/build/elm/elm-tmp/Reporting/Error/Import.dyn_o )
[ 49 of 132] Compiling Generate.JavaScript.Name ( compiler/src/Generate/JavaScript/Name.hs, dist/build/elm/elm-tmp/Generate/JavaScript/Name.o, dist/build/elm/elm-tmp/Generate/JavaScript/Name.dyn_o )

compiler/src/Generate/JavaScript/Name.hs:22:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
22 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 50 of 132] Compiling Generate.JavaScript.Builder ( compiler/src/Generate/JavaScript/Builder.hs, dist/build/elm/elm-tmp/Generate/JavaScript/Builder.o, dist/build/elm/elm-tmp/Generate/JavaScript/Builder.dyn_o )

compiler/src/Generate/JavaScript/Builder.hs:21:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
21 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 51 of 132] Compiling Elm.Compiler.Imports ( compiler/src/Elm/Compiler/Imports.hs, dist/build/elm/elm-tmp/Elm/Compiler/Imports.o, dist/build/elm/elm-tmp/Elm/Compiler/Imports.dyn_o )
[ 52 of 132] Compiling Parse.Module     ( compiler/src/Parse/Module.hs, dist/build/elm/elm-tmp/Parse/Module.o, dist/build/elm/elm-tmp/Parse/Module.dyn_o )
[ 53 of 132] Compiling AST.Canonical    ( compiler/src/AST/Canonical.hs, dist/build/elm/elm-tmp/AST/Canonical.o, dist/build/elm/elm-tmp/AST/Canonical.dyn_o )
[ 54 of 132] Compiling Reporting.Render.Type ( compiler/src/Reporting/Render/Type.hs, dist/build/elm/elm-tmp/Reporting/Render/Type.o, dist/build/elm/elm-tmp/Reporting/Render/Type.dyn_o )

compiler/src/Reporting/Render/Type.hs:24:36: warning: [-Wunused-imports]
    The import of ‘<>’ from module ‘Reporting.Doc’ is redundant
   |
24 | import Reporting.Doc ( Doc, (<+>), (<>) )
   |                                    ^^^^
[ 55 of 132] Compiling Elm.Compiler.Type ( compiler/src/Elm/Compiler/Type.hs, dist/build/elm/elm-tmp/Elm/Compiler/Type.o, dist/build/elm/elm-tmp/Elm/Compiler/Type.dyn_o )

compiler/src/Elm/Compiler/Type.hs:66:9: warning: [GHC-62161] [-Wincomplete-uni-patterns]
    Pattern match(es) are non-exhaustive
    In a pattern binding:
        Patterns of type ‘[D.Doc]’ not matched:
            []
            [_]
   |
66 |         a:b:cs =
   |         ^^^^^^^^...
[ 56 of 132] Compiling Reporting.Error.Canonicalize ( compiler/src/Reporting/Error/Canonicalize.hs, dist/build/elm/elm-tmp/Reporting/Error/Canonicalize.o, dist/build/elm/elm-tmp/Reporting/Error/Canonicalize.dyn_o )

compiler/src/Reporting/Error/Canonicalize.hs:29:35: warning: [-Wunused-imports]
    The import of ‘<>’ from module ‘Reporting.Doc’ is redundant
   |
29 | import Reporting.Doc (Doc, (<+>), (<>))
   |                                   ^^^^
[ 57 of 132] Compiling Reporting.Error.Main ( compiler/src/Reporting/Error/Main.hs, dist/build/elm/elm-tmp/Reporting/Error/Main.o, dist/build/elm/elm-tmp/Reporting/Error/Main.dyn_o )
[ 58 of 132] Compiling Optimize.DecisionTree ( compiler/src/Optimize/DecisionTree.hs, dist/build/elm/elm-tmp/Optimize/DecisionTree.o, dist/build/elm/elm-tmp/Optimize/DecisionTree.dyn_o )
[ 59 of 132] Compiling Nitpick.PatternMatches ( compiler/src/Nitpick/PatternMatches.hs, dist/build/elm/elm-tmp/Nitpick/PatternMatches.o, dist/build/elm/elm-tmp/Nitpick/PatternMatches.dyn_o )
[ 60 of 132] Compiling Reporting.Error.Pattern ( compiler/src/Reporting/Error/Pattern.hs, dist/build/elm/elm-tmp/Reporting/Error/Pattern.o, dist/build/elm/elm-tmp/Reporting/Error/Pattern.dyn_o )

compiler/src/Reporting/Error/Pattern.hs:13:1: warning: [-Wunused-imports]
    The import of ‘Reporting.Doc’ is redundant
      except perhaps to import instances from ‘Reporting.Doc’
    To import instances alone, use: import Reporting.Doc()
   |
13 | import Reporting.Doc ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 61 of 132] Compiling AST.Utils.Type   ( compiler/src/AST/Utils/Type.hs, dist/build/elm/elm-tmp/AST/Utils/Type.o, dist/build/elm/elm-tmp/AST/Utils/Type.dyn_o )
[ 62 of 132] Compiling Elm.Kernel       ( compiler/src/Elm/Kernel.hs, dist/build/elm/elm-tmp/Elm/Kernel.o, dist/build/elm/elm-tmp/Elm/Kernel.dyn_o )
[ 63 of 132] Compiling Elm.Interface    ( compiler/src/Elm/Interface.hs, dist/build/elm/elm-tmp/Elm/Interface.o, dist/build/elm/elm-tmp/Elm/Interface.dyn_o )
[ 64 of 132] Compiling Deps.Website     ( builder/src/Deps/Website.hs, dist/build/elm/elm-tmp/Deps/Website.o, dist/build/elm/elm-tmp/Deps/Website.dyn_o )
[ 65 of 132] Compiling AST.Optimized    ( compiler/src/AST/Optimized.hs, dist/build/elm/elm-tmp/AST/Optimized.o, dist/build/elm/elm-tmp/AST/Optimized.dyn_o )
[ 66 of 132] Compiling Optimize.Names   ( compiler/src/Optimize/Names.hs, dist/build/elm/elm-tmp/Optimize/Names.o, dist/build/elm/elm-tmp/Optimize/Names.dyn_o )
[ 67 of 132] Compiling Optimize.Port    ( compiler/src/Optimize/Port.hs, dist/build/elm/elm-tmp/Optimize/Port.o, dist/build/elm/elm-tmp/Optimize/Port.dyn_o )
[ 68 of 132] Compiling Optimize.Case    ( compiler/src/Optimize/Case.hs, dist/build/elm/elm-tmp/Optimize/Case.o, dist/build/elm/elm-tmp/Optimize/Case.dyn_o )
[ 69 of 132] Compiling Optimize.Expression ( compiler/src/Optimize/Expression.hs, dist/build/elm/elm-tmp/Optimize/Expression.o, dist/build/elm/elm-tmp/Optimize/Expression.dyn_o )
[ 70 of 132] Compiling Nitpick.Debug    ( compiler/src/Nitpick/Debug.hs, dist/build/elm/elm-tmp/Nitpick/Debug.o, dist/build/elm/elm-tmp/Nitpick/Debug.dyn_o )
[ 71 of 132] Compiling Elm.Compiler.Type.Extract ( compiler/src/Elm/Compiler/Type/Extract.hs, dist/build/elm/elm-tmp/Elm/Compiler/Type/Extract.o, dist/build/elm/elm-tmp/Elm/Compiler/Type/Extract.dyn_o )
[ 72 of 132] Compiling Generate.Mode    ( compiler/src/Generate/Mode.hs, dist/build/elm/elm-tmp/Generate/Mode.o, dist/build/elm/elm-tmp/Generate/Mode.dyn_o )
[ 73 of 132] Compiling Generate.JavaScript.Expression ( compiler/src/Generate/JavaScript/Expression.hs, dist/build/elm/elm-tmp/Generate/JavaScript/Expression.o, dist/build/elm/elm-tmp/Generate/JavaScript/Expression.dyn_o )
[ 74 of 132] Compiling Generate.JavaScript ( compiler/src/Generate/JavaScript.hs, dist/build/elm/elm-tmp/Generate/JavaScript.o, dist/build/elm/elm-tmp/Generate/JavaScript.dyn_o )
[ 75 of 132] Compiling Elm.Licenses     ( compiler/src/Elm/Licenses.hs, dist/build/elm/elm-tmp/Elm/Licenses.o, dist/build/elm/elm-tmp/Elm/Licenses.dyn_o )
[ 76 of 132] Compiling Reporting.Task   ( builder/src/Reporting/Task.hs, dist/build/elm/elm-tmp/Reporting/Task.o, dist/build/elm/elm-tmp/Reporting/Task.dyn_o )
[ 77 of 132] Compiling Reporting.Warning ( compiler/src/Reporting/Warning.hs, dist/build/elm/elm-tmp/Reporting/Warning.o, dist/build/elm/elm-tmp/Reporting/Warning.dyn_o )

compiler/src/Reporting/Warning.hs:11:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
11 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 78 of 132] Compiling Reporting.Result ( compiler/src/Reporting/Result.hs, dist/build/elm/elm-tmp/Reporting/Result.o, dist/build/elm/elm-tmp/Reporting/Result.dyn_o )

compiler/src/Reporting/Result.hs:110:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘return’ definition detected
    in the instance declaration for ‘Monad (Result i w e)’.
    ‘return’ will eventually be removed in favour of ‘pure’
    Either remove definition for ‘return’ (recommended) or define as ‘return = pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
110 |   return = ok
    |   ^^^^^^^^^^^

compiler/src/Reporting/Result.hs:121:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘(>>)’ definition detected
    in the instance declaration for ‘Monad (Result i w e)’.
    ‘(>>)’ will eventually be removed in favour of ‘(*>)’
    Either remove definition for ‘(>>)’ (recommended) or define as ‘(>>) = (*>)’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
121 |   (>>) (Result ka) (Result kb) =
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
[ 79 of 132] Compiling Elm.Docs         ( compiler/src/Elm/Docs.hs, dist/build/elm/elm-tmp/Elm/Docs.o, dist/build/elm/elm-tmp/Elm/Docs.dyn_o )
[ 80 of 132] Compiling Canonicalize.Environment.Dups ( compiler/src/Canonicalize/Environment/Dups.hs, dist/build/elm/elm-tmp/Canonicalize/Environment/Dups.o, dist/build/elm/elm-tmp/Canonicalize/Environment/Dups.dyn_o )
[ 81 of 132] Compiling Canonicalize.Environment ( compiler/src/Canonicalize/Environment.hs, dist/build/elm/elm-tmp/Canonicalize/Environment.o, dist/build/elm/elm-tmp/Canonicalize/Environment.dyn_o )
[ 82 of 132] Compiling Canonicalize.Type ( compiler/src/Canonicalize/Type.hs, dist/build/elm/elm-tmp/Canonicalize/Type.o, dist/build/elm/elm-tmp/Canonicalize/Type.dyn_o )
[ 83 of 132] Compiling Canonicalize.Pattern ( compiler/src/Canonicalize/Pattern.hs, dist/build/elm/elm-tmp/Canonicalize/Pattern.o, dist/build/elm/elm-tmp/Canonicalize/Pattern.dyn_o )
[ 84 of 132] Compiling Canonicalize.Environment.Local ( compiler/src/Canonicalize/Environment/Local.hs, dist/build/elm/elm-tmp/Canonicalize/Environment/Local.o, dist/build/elm/elm-tmp/Canonicalize/Environment/Local.dyn_o )
[ 85 of 132] Compiling Canonicalize.Environment.Foreign ( compiler/src/Canonicalize/Environment/Foreign.hs, dist/build/elm/elm-tmp/Canonicalize/Environment/Foreign.o, dist/build/elm/elm-tmp/Canonicalize/Environment/Foreign.dyn_o )
[ 86 of 132] Compiling Canonicalize.Effects ( compiler/src/Canonicalize/Effects.hs, dist/build/elm/elm-tmp/Canonicalize/Effects.o, dist/build/elm/elm-tmp/Canonicalize/Effects.dyn_o )
[ 87 of 132] Compiling Optimize.Module  ( compiler/src/Optimize/Module.hs, dist/build/elm/elm-tmp/Optimize/Module.o, dist/build/elm/elm-tmp/Optimize/Module.dyn_o )
[ 88 of 132] Compiling Canonicalize.Expression ( compiler/src/Canonicalize/Expression.hs, dist/build/elm/elm-tmp/Canonicalize/Expression.o, dist/build/elm/elm-tmp/Canonicalize/Expression.dyn_o )
[ 89 of 132] Compiling Canonicalize.Module ( compiler/src/Canonicalize/Module.hs, dist/build/elm/elm-tmp/Canonicalize/Module.o, dist/build/elm/elm-tmp/Canonicalize/Module.dyn_o )
[ 90 of 132] Compiling Stuff            ( builder/src/Stuff.hs, dist/build/elm/elm-tmp/Stuff.o, dist/build/elm/elm-tmp/Stuff.dyn_o )
[ 91 of 132] Compiling Terminal.Internal ( terminal/impl/Terminal/Internal.hs, dist/build/elm/elm-tmp/Terminal/Internal.o, dist/build/elm/elm-tmp/Terminal/Internal.dyn_o )
[ 92 of 132] Compiling Terminal.Error   ( terminal/impl/Terminal/Error.hs, dist/build/elm/elm-tmp/Terminal/Error.o, dist/build/elm/elm-tmp/Terminal/Error.dyn_o )
[ 93 of 132] Compiling Terminal.Chomp   ( terminal/impl/Terminal/Chomp.hs, dist/build/elm/elm-tmp/Terminal/Chomp.o, dist/build/elm/elm-tmp/Terminal/Chomp.dyn_o )
[ 94 of 132] Compiling Terminal         ( terminal/impl/Terminal.hs, dist/build/elm/elm-tmp/Terminal.o, dist/build/elm/elm-tmp/Terminal.dyn_o )
[ 95 of 132] Compiling Type.Error       ( compiler/src/Type/Error.hs, dist/build/elm/elm-tmp/Type/Error.o, dist/build/elm/elm-tmp/Type/Error.dyn_o )

compiler/src/Type/Error.hs:23:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
23 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 96 of 132] Compiling Reporting.Error.Type ( compiler/src/Reporting/Error/Type.hs, dist/build/elm/elm-tmp/Reporting/Error/Type.o, dist/build/elm/elm-tmp/Reporting/Error/Type.dyn_o )

compiler/src/Reporting/Error/Type.hs:24:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
24 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
[ 97 of 132] Compiling Reporting.Error  ( compiler/src/Reporting/Error.hs, dist/build/elm/elm-tmp/Reporting/Error.o, dist/build/elm/elm-tmp/Reporting/Error.dyn_o )
[ 98 of 132] Compiling Reporting.Exit.Help ( builder/src/Reporting/Exit/Help.hs, dist/build/elm/elm-tmp/Reporting/Exit/Help.o, dist/build/elm/elm-tmp/Reporting/Exit/Help.dyn_o )
[ 99 of 132] Compiling Reporting.Error.Json ( compiler/src/Reporting/Error/Json.hs, dist/build/elm/elm-tmp/Reporting/Error/Json.o, dist/build/elm/elm-tmp/Reporting/Error/Json.dyn_o )
[100 of 132] Compiling Reporting.Exit   ( builder/src/Reporting/Exit.hs, dist/build/elm/elm-tmp/Reporting/Exit.o, dist/build/elm/elm-tmp/Reporting/Exit.dyn_o )
[101 of 132] Compiling Reporting        ( builder/src/Reporting.hs, dist/build/elm/elm-tmp/Reporting.o, dist/build/elm/elm-tmp/Reporting.dyn_o )
[102 of 132] Compiling Elm.Outline      ( builder/src/Elm/Outline.hs, dist/build/elm/elm-tmp/Elm/Outline.o, dist/build/elm/elm-tmp/Elm/Outline.dyn_o )
[103 of 132] Compiling Deps.Registry    ( builder/src/Deps/Registry.hs, dist/build/elm/elm-tmp/Deps/Registry.o, dist/build/elm/elm-tmp/Deps/Registry.dyn_o )
[104 of 132] Compiling Terminal.Helpers ( terminal/impl/Terminal/Helpers.hs, dist/build/elm/elm-tmp/Terminal/Helpers.o, dist/build/elm/elm-tmp/Terminal/Helpers.dyn_o )
[105 of 132] Compiling Deps.Solver      ( builder/src/Deps/Solver.hs, dist/build/elm/elm-tmp/Deps/Solver.o, dist/build/elm/elm-tmp/Deps/Solver.dyn_o )

builder/src/Deps/Solver.hs:411:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘return’ definition detected
    in the instance declaration for ‘Monad Solver’.
    ‘return’ will eventually be removed in favour of ‘pure’
    Either remove definition for ‘return’ (recommended) or define as ‘return = pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
411 |   return a =
    |   ^^^^^^^^^^...
[106 of 132] Compiling Init             ( terminal/src/Init.hs, dist/build/elm/elm-tmp/Init.o, dist/build/elm/elm-tmp/Init.dyn_o )
[107 of 132] Compiling Deps.Bump        ( builder/src/Deps/Bump.hs, dist/build/elm/elm-tmp/Deps/Bump.o, dist/build/elm/elm-tmp/Deps/Bump.dyn_o )
[108 of 132] Compiling Deps.Diff        ( builder/src/Deps/Diff.hs, dist/build/elm/elm-tmp/Deps/Diff.o, dist/build/elm/elm-tmp/Deps/Diff.dyn_o )
[109 of 132] Compiling Type.UnionFind   ( compiler/src/Type/UnionFind.hs, dist/build/elm/elm-tmp/Type/UnionFind.o, dist/build/elm/elm-tmp/Type/UnionFind.dyn_o )
[110 of 132] Compiling Type.Type        ( compiler/src/Type/Type.hs, dist/build/elm/elm-tmp/Type/Type.o, dist/build/elm/elm-tmp/Type/Type.dyn_o )
[111 of 132] Compiling Type.Instantiate ( compiler/src/Type/Instantiate.hs, dist/build/elm/elm-tmp/Type/Instantiate.o, dist/build/elm/elm-tmp/Type/Instantiate.dyn_o )
[112 of 132] Compiling Type.Constrain.Pattern ( compiler/src/Type/Constrain/Pattern.hs, dist/build/elm/elm-tmp/Type/Constrain/Pattern.o, dist/build/elm/elm-tmp/Type/Constrain/Pattern.dyn_o )
[113 of 132] Compiling Type.Constrain.Expression ( compiler/src/Type/Constrain/Expression.hs, dist/build/elm/elm-tmp/Type/Constrain/Expression.o, dist/build/elm/elm-tmp/Type/Constrain/Expression.dyn_o )
[114 of 132] Compiling Type.Constrain.Module ( compiler/src/Type/Constrain/Module.hs, dist/build/elm/elm-tmp/Type/Constrain/Module.o, dist/build/elm/elm-tmp/Type/Constrain/Module.dyn_o )
[115 of 132] Compiling Type.Occurs      ( compiler/src/Type/Occurs.hs, dist/build/elm/elm-tmp/Type/Occurs.o, dist/build/elm/elm-tmp/Type/Occurs.dyn_o )
[116 of 132] Compiling Type.Unify       ( compiler/src/Type/Unify.hs, dist/build/elm/elm-tmp/Type/Unify.o, dist/build/elm/elm-tmp/Type/Unify.dyn_o )

compiler/src/Type/Unify.hs:93:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘return’ definition detected
    in the instance declaration for ‘Monad Unify’.
    ‘return’ will eventually be removed in favour of ‘pure’
    Either remove definition for ‘return’ (recommended) or define as ‘return = pure’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
   |
93 |   return a =
   |   ^^^^^^^^^^...

compiler/src/Type/Unify.hs:106:3: warning: [-Wnoncanonical-monad-instances]
    Noncanonical ‘(>>)’ definition detected
    in the instance declaration for ‘Monad Unify’.
    ‘(>>)’ will eventually be removed in favour of ‘(*>)’
    Either remove definition for ‘(>>)’ (recommended) or define as ‘(>>) = (*>)’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
    |
106 |   (>>) (Unify ka) (Unify kb) =
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
[117 of 132] Compiling Type.Solve       ( compiler/src/Type/Solve.hs, dist/build/elm/elm-tmp/Type/Solve.o, dist/build/elm/elm-tmp/Type/Solve.dyn_o )
[118 of 132] Compiling Compile          ( compiler/src/Compile.hs, dist/build/elm/elm-tmp/Compile.o, dist/build/elm/elm-tmp/Compile.dyn_o )
[119 of 132] Compiling Elm.Details      ( builder/src/Elm/Details.hs, dist/build/elm/elm-tmp/Elm/Details.o, dist/build/elm/elm-tmp/Elm/Details.dyn_o )
[120 of 132] Compiling Install          ( terminal/src/Install.hs, dist/build/elm/elm-tmp/Install.o, dist/build/elm/elm-tmp/Install.dyn_o )
[121 of 132] Compiling Develop.Generate.Index ( terminal/src/Develop/Generate/Index.hs, dist/build/elm/elm-tmp/Develop/Generate/Index.o, dist/build/elm/elm-tmp/Develop/Generate/Index.dyn_o )
[122 of 132] Compiling Build            ( builder/src/Build.hs, dist/build/elm/elm-tmp/Build.o, dist/build/elm/elm-tmp/Build.dyn_o )
[123 of 132] Compiling Publish          ( terminal/src/Publish.hs, dist/build/elm/elm-tmp/Publish.o, dist/build/elm/elm-tmp/Publish.dyn_o )
[124 of 132] Compiling Generate         ( builder/src/Generate.hs, dist/build/elm/elm-tmp/Generate.o, dist/build/elm/elm-tmp/Generate.dyn_o )
[125 of 132] Compiling Repl             ( terminal/src/Repl.hs, dist/build/elm/elm-tmp/Repl.o, dist/build/elm/elm-tmp/Repl.dyn_o )

terminal/src/Repl.hs:32:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
32 | import Data.Monoid ((<>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

terminal/src/Repl.hs:530:42: warning: [GHC-62161] [-Wincomplete-uni-patterns]
    Pattern match(es) are non-exhaustive
    In a lambda abstraction:
        Patterns of type ‘Maybe IO.Handle’, ‘Maybe IO.Handle’,
                         ‘Maybe IO.Handle’, ‘Proc.ProcessHandle’ not matched:
            Nothing _ _ _
    |
530 |   Proc.withCreateProcess createProcess $ \(Just stdin) _ _ handle ->
    |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^...
[126 of 132] Compiling Make             ( terminal/src/Make.hs, dist/build/elm/elm-tmp/Make.o, dist/build/elm/elm-tmp/Make.dyn_o )
[127 of 132] Compiling Diff             ( terminal/src/Diff.hs, dist/build/elm/elm-tmp/Diff.o, dist/build/elm/elm-tmp/Diff.dyn_o )

terminal/src/Diff.hs:30:23: warning: [-Wunused-imports]
    The import of ‘<>’ from module ‘Reporting.Doc’ is redundant
   |
30 | import Reporting.Doc ((<>), (<+>))
   |                       ^^^^
[128 of 132] Compiling Develop.StaticFiles.Build ( terminal/src/Develop/StaticFiles/Build.hs, dist/build/elm/elm-tmp/Develop/StaticFiles/Build.o, dist/build/elm/elm-tmp/Develop/StaticFiles/Build.dyn_o )
[129 of 132] Compiling Develop.StaticFiles ( terminal/src/Develop/StaticFiles.hs, dist/build/elm/elm-tmp/Develop/StaticFiles.o, dist/build/elm/elm-tmp/Develop/StaticFiles.dyn_o )
ghc-9.6.4: security: createProcess: posix_spawnp: does not exist (No such file or directory)
```


### Notify maintainers

@domenkozar @turboMaCk 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.3.0, macOS 14.3.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.19.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `.../.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/haskell-modules/configuration-darwin.nix
+++ b/pkgs/development/haskell-modules/configuration-darwin.nix
@@ -110,6 +110,12 @@ self: super: ({
         substituteInPlace System/X509/MacOS.hs --replace security /usr/bin/security
       '' + (drv.postPatch or "");
     }) super.x509-system;
+  crypton-x509-system = overrideCabal (drv:
+    lib.optionalAttrs (!pkgs.stdenv.cc.nativeLibc) {
+      postPatch = ''
+        substituteInPlace System/X509/MacOS.hs --replace security /usr/bin/security
+      '' + (drv.postPatch or "");
+    }) super.crypton-x509-system;
 
   # https://github.com/haskell-foundation/foundation/pull/412
   foundation = dontCheck super.foundation;

